vendor/github.com/ethereum/go-ethereum/les/handler.go

```diff
@@ -210,9 +215,31 @@ func NewProtocolManager(chainConfig *params.ChainConfig, lightSync bool, protoco
 		manager.fetcher = newLightFetcher(manager)
 	}
 
+	go manager.monitorDownloads()
+
 	return manager, nil
 }
 
+func (pm *ProtocolManager) monitorDownloads() {
+	sub := pm.eventMux.Subscribe(downloader.StartEvent{}, downloader.DoneEvent{}, downloader.FailedEvent{})
```

Comment:
is it possible to wait for all goroutines to finish in downloader and export blocking api such as Stop?

i know that downloader have Terminate and Cancel methods but they indeed don't affect rollback and there is no interface to wait until all downloader goroutines are finished

Reply:
I checked ‘eth’ and they don’t use rollback, so it’s very unlikely that it will happen there.
I was thinking about either making Terminate blocking or returning a WaitGroup from there. What do you think?

Reply:
Should we unsubscribe finally? Or when `sub.Chan()` is closed it's enough?

Reply:
Good point, I think it is cleaner to unsubscribe.

Reply:
@mandrigin making Terminate blocking sounds good, if it doesn't conflict with another downloader usages

Reply:
I'll do that for the upstream PR then.